### PR TITLE
Ticket 7 - Submitted For Review

### PR DESCRIPTION
**Pull request by kieron-day**
Ticket 7 - PATCH /api/articles/:article_id

Created Patch request using TDD
Added patchArticleById functionality to controller
Added updateArticleById functionality to model

Happy Path - 200 - responds with correct updated object
Sad Path - 400 - Invalid request body used (e.g. String as property)
Sad Path 404 - path not found (handled in previous ticket)
Sad Path 400 - invalid path type (string) - (handled in previous ticket)
Sad path 404 - valid but not present article id (handled in previous ticket)

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -75,4 +75,45 @@ describe("app", () => {
...
+		});
+		test("Status 400 - Invalid request body used for patch", () => {
+			const body = { inc_votes: "Not a number" };
+			return request(app)
+				.patch("/api/articles/1")
+				.send(body)
+				.expect(400)
+				.then(({ body: { message } }) => {
+					expect(message).toBe("Bad Request");
+				});
```
Comment (Angaca) [inline __tests__/app.test.js]: Same test twice?

File: controllers/articles.controllers.js [hunk 2]
```diff
@@ -10,3 +13,15 @@ exports.getArticleById = (req, res, next) => {
 			next(err);
 		});
 };
+
+exports.patchArticleById = (req, res, next) => {
+	const articleId = req.params.article_id;
+	const incVotes = req.body.inc_votes;
+	updateArticleById(articleId, incVotes)
+		.then((updatedArticle) => {
+			res.status(200).send({ updatedArticle });
```
Comment (Angaca) [inline controllers/articles.controllers.js]: Should still live in a key of article

**Review by Angaca — APPROVED** [review]
Good to go!

File: controllers/articles.controllers.js (hunk 2 above)
Comment (kieron-day) [reply controllers/articles.controllers.js]: Ah, makes sense for future use - Thanks Andrea

**Pull request merged**